integritybackend/actions.py

```diff
@@ -53,12 +54,13 @@ def archive(self, asset_fullpath: str, org_config: dict, collection_id: str):
 
         # Pull things out of the config
         collection_config = next(
-            (c for c in org_config["collections"] if c.get("id") == collection_id), None
+            (c for c in org_config["collections"] if org_config["collections"][c]['conf']['id'] == collection_id), None
```

Comment:
c is a string so `get("id")` failed.

Reply:
The commited code works

This is my understanding of why the above does not work (non of the code below was tested)

`for c in org_config["collections"]` iterates through `keys` of `org_config["collections"]`  
`.values()` i think would return values

IE:
`(c for c in org_config["collections"].values() if c.get("id") == collection_id), None`

However i think this would still cause an issue since `id` is not a member of the resulting `c` dict.  its `collections->[KEY]->conf->id` where `c = [key] `

so maybe `c.get("conf").get("id")` ?

Reply:
```suggestion
            (c for c in org_config["collections"] if org_config["collections"][c]["conf"["id"] == collection_id), None
```
consistent quotes.

Reply:
I did this little test with the original code, and it works in the way I expect it to:

```
import json

collection_id = "test-web-archive"
org_config = json.loads('{"organizations":[{"id":"starling-lab-test","collections":[{"id":"test-web-archive","asset_extensions":["wacz"],"actions":[]}]}]}')["organizations"][0]
collection_config = next(
    (c for c in org_config["collections"] if c.get("id") == collection_id), None
)
print(collection_config)

# ❯❯❯ python play.py
# {u'asset_extensions': [u'wacz'], u'id': u'test-web-archive', u'actions': []}
```

Am I missing something here? Perhaps the client is passing the wrong thing?

Where does `conf` come from? That is not a key we have. Is your config.json right?

Reply:
This is the output of `org_config` i get before the collection_config = line

```
{
    "id": "starling-lab-test",
    "collections": {
        "test-bot-archive": {
            "conf": {
                "id": "test-bot-archive",
                "asset_extensions": [
                    "zip"
                ],
                "actions": [
                    {
                        "name": "archive",
                        "params": {
                            "authsigner": "starlinglab-authsign",
                            "encryption": {
                                "algo": "aes-256-cbc",
                                "key": "starlinglab-aes-256"
                            },
                            "registration_policies": {
                                "opentimestamps": {
                                    "active": True
                                },
                                "iscn": {
                                    "active": True
                                },
                                "numbersprotocol": {
                                    "active": True
                                }
                            }
                        }
                    }
                ]
            },
            "actions": {
                "archive": {
                    "name": "archive",
                    "params": {
                        "authsigner": "starlinglab-authsign",
                        "encryption": {
                            "algo": "aes-256-cbc",
                            "key": "starlinglab-aes-256"
                        },
                        "registration_policies": {
                            "opentimestamps": {
                                "active": True
                            },
                            "iscn": {
                                "active": True
                            },
                            "numbersprotocol": {
                                "active": True
                            }
                        }
                    }
                }
            }
        }
    }
}
```

Reply:
"conf" object set on https://github.com/starlinglab/integrity-backend/blob/patches/integritybackend/config.py#L89

Reply:
> I did this little test with the original code, and it works in the way I expect it to:

Yes code worked for me to

I think the difference is that in your input json and the JSON I see are differnt
specifically `collections` in yours is an array `[]` not an object `{}`  
Since arrays don't have a key you get the value back

The JSON i get back collections is an object `{}` so the key is returned

Reply:
config.py re-written to use underlying json layer.